packages/core/content-releases/server/src/controllers/release.ts

```diff
@@ -29,50 +33,110 @@ const releaseController = {
 
     const releaseService = getService('release', { strapi });
 
-    // Handle requests for releases filtered by content type entry
-    const isFindManyForContentTypeEntry = Boolean(ctx.query?.contentTypeUid && ctx.query?.entryId);
-    if (isFindManyForContentTypeEntry) {
-      const query: GetContentTypeEntryReleases.Request['query'] =
-        await permissionsManager.sanitizeQuery(ctx.query);
+    const query = await permissionsManager.sanitizeQuery(ctx.query);
+    const { contentTypeUid, documentId, hasEntryAttached, locale } = query;
+    const entryId = await getEntryId({ contentTypeUid, documentId, locale }, { strapi });
 
-      const contentTypeUid = query.contentTypeUid;
-      const entryId = query.entryId;
-      // Parse the string value or fallback to a default
-      const hasEntryAttached: GetContentTypeEntryReleases.Request['query']['hasEntryAttached'] =
-        typeof query.hasEntryAttached === 'string' ? JSON.parse(query.hasEntryAttached) : false;
-
-      const data = hasEntryAttached
-        ? await releaseService.findManyWithContentTypeEntryAttached(contentTypeUid, entryId)
-        : await releaseService.findManyWithoutContentTypeEntryAttached(contentTypeUid, entryId);
-
-      ctx.body = { data };
-    } else {
-      const query: GetReleases.Request['query'] = await permissionsManager.sanitizeQuery(ctx.query);
-      const { results, pagination } = await releaseService.findPage(query);
+    if (!entryId) {
+      throw new errors.NotFoundError('Entry not found');
+    }
 
-      const data = results.map((release: ReleaseWithPopulatedActions) => {
-        const { actions, ...releaseData } = release;
+    const isEntryAttachedBoolean =
```

Comment:
Personally `isXXX` convention is sufficient for indicating a boolean, so the Boolean part of this variable name seems redundant. 

The type that comes through ends up being any... Maybe we could wrap JSON.parse in Boolean 🤷‍♂️